opengemini-client-common/src/main/java/io/opengemini/client/common/BaseClient.java

```diff
@@ -24,12 +40,55 @@ public BaseClient(BaseConfiguration conf) {
         }
         for (Address address : conf.getAddresses()) {
             this.serverUrls.add(httpPrefix + address.getHost() + ":" + address.getPort());
+            this.serverUrlsWork.add(new AtomicBoolean(true));
         }
+        startHealthCheck();
+    }
+
+    // start health check
+    private void startHealthCheck() {
+        scheduler.scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+                for (String url : serverUrls) {
+                    try {
+                        HttpRequest request = HttpRequest.newBuilder()
+                                .uri(URI.create(url + getPingUrl()))
+                                .GET()
+                                .timeout(Duration.ofSeconds(2))
+                                .build();
+                        HttpResponse<String> response = healthCheckClient
+                                .sendAsync(request, HttpResponse.BodyHandlers.ofString()).get();
+                        System.out.println(response.statusCode());
+                        if (response.statusCode() >= 200 && response.statusCode() < 300) {
+                            serverUrlsWork.set(serverUrls.indexOf(url), new AtomicBoolean(true));
```

Comment:
Creating a new object and performing a modification action for each period is unreasonable. We should only modify the class when the state changes.